File: 2023/day1.py
import re

with open("_input/day1.txt", encoding='utf8') as f:
    lines = f.read().splitlines()

def trans(num):
    if num == 'one': return 1
    if num == 'two': return 2
    if num == 'three': return 3
    if num == 'four': return 4
    if num == 'five': return 5
    if num == 'six': return 6
    if num == 'seven': return 7
    if num == 'eight': return 8
    if num == 'nine': return 9
    if num.isnumeric():
        return num

t = 0
for line in lines:
    res = re.findall('(?=(\d|one|two|three|four|five|six|seven|eight|nine))', line)
    c = int(f'{trans(res[0])}{trans(res[-1])}')
    t += c
print(t)

File: 2023/go/day1.go
package main

import (
	"aoc/2023/utils"
	"fmt"
	"regexp"
)

func translate(num string) int {
	switch num {
	case "one":
		return 1
	case "two":
		return 2
	case "three":
		return 3
	case "four":
		return 4
	case "five":
		return 5
	case "six":
		return 6
	case "seven":
		return 7
	case "eight":
		return 8
	case "nine":
		return 9
	default:
		return 0 // Return 0 if the input string is not a number word
	}
}

func main() {
	part1 := 0
	regex := regexp.MustCompile(`(\d|one|two|three|four|five|six|seven|eight|nine)`)
	for _, line := range utils.GetFileLines("../_input/day1.txt") {
		matches := regex.FindAllString(line, -1)
		part1 += translate(matches[0]) + translate(matches[len(matches)-1])
	}
	fmt.Println(part1)
}
